terragrunt/org_account/spend_notifier/lambdas/spend_notifier/spend_notifier.js

```diff
@@ -0,0 +1,173 @@
+const AWS = require('aws-sdk');
+const costexplorer = new AWS.CostExplorer({ region: 'us-east-1' });
+const organizations = new AWS.Organizations({ region: 'us-east-1' });
+
+const https = require('https')
+
+exports.handler = async (event) => {
+  const hook = event.hook;
+  const today = new Date();
+  const accountCost = await getAccountCost()
+  let accounts = await getAccounts()
+  Object.keys(accounts).forEach(key => {
+    if(accountCost.hasOwnProperty(key)){
+      accounts[key]["Cost"] = accountCost[key]
+    } else {
+      accounts[key]["Cost"] = 0
+    }
+  })
+  let BU = {}
+
+  Object.values(accounts).forEach(account => {
+    if (BU.hasOwnProperty(account["BU"])) {
+      BU[account["BU"]] = BU[account["BU"]] + account["Cost"]
+    } else {
+      BU[account["BU"]] = account["Cost"]
+    }
+  })
+  const totalCost = Object.values(BU).reduce((a, b) => a + b, 0)
+
+  const data = JSON.stringify(
+    {
+      "blocks": [
+        {
+          "type": "section",
+          "text": {
+            "type": "plain_text",
+            "text": `Current AWS spend for ${(today.getMonth() + 1).pad(2)}-${today.getFullYear()}`,
+            "emoji": true
+          }
+        },
+        {
+          "type": "section",
+          "fields": Object.entries(BU).map(bu => (
+            [{ "type": "mrkdwn", "text": `*${bu[0]}*` }, { "type": "mrkdwn", "text": `$${bu[1].toFixed(2).replace(/\d(?=(\d{3})+\.)/g, '$&,')} USD` }]
+          )).flat()
+        },
+        {
+          "type": "divider"
+        },
+        {
+          "type": "section",
+          "fields": [{ "type": "mrkdwn", "text": `*Total*` }, { "type": "mrkdwn", "text": `$${totalCost.toFixed(2).replace(/\d(?=(\d{3})+\.)/g, '$&,')} USD` }]
+        },
+        /*
+        {
+          "type": "actions",
+          "elements": {
+            "action_id": "text1234",
+            "type": "static_select",
+            "placeholder": {
+              "type": "plain_text",
+              "text": "Click for more details"
+            },
+            "options": Object.values(accounts).map(account => {
+             return {
+                "text": {
+                  "type": "plain_text",
+                  "text": `${account["Name"]} (${account["BU"]}): $${account["Cost"].toFixed(2).replace(/\d(?=(\d{3})+\.)/g, '$&,')} USD`,
+                },
+                "value": account["Name"]
+              } 
+            }),
+            "action_id": "overflow-action"
+          }
+        }
+        */
```

Comment:
Can we remove this commented out block?

Reply:
yeah not a problem that was left over from the code Max wrote didn't even review the lambda itself to be honest.